src/TensorOneD.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+
+#include "config.h"
+
+#include "TensorOneD.hpp"
+
+#include <math.h>
+#include <algorithm>
+#include <functional>
+#include <numeric>
+#include <utility>
+
+#include "geopm/Exception.hpp"
+
+namespace geopm
+{
+    TensorOneD::TensorOneD()
+    {
+    }
+
+    TensorOneD::TensorOneD(std::size_t dim)
+    {
+        set_dim(dim);
+    }
+
+    TensorOneD::TensorOneD(const TensorOneD &other)
+        : m_vec(other.m_vec)
+    {
+    }
+
+    TensorOneD::TensorOneD(TensorOneD &&other)
+        : m_vec(std::move(other.m_vec))
+    {
+    }
+
+    TensorOneD::TensorOneD(json11::Json input)
+    {
+        if (!input.is_array()) {
+            throw geopm::Exception("Neural network weights is non-array-type.\n",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+        if (input.array_items().size() == 0) {
+            throw geopm::Exception("Empty array is invalid for neural network weights.\n",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        set_dim(input.array_items().size());
+
+        for (std::size_t idx = 0; idx < m_vec.size(); idx++) {
```

Comment:
You can pull the `.size()` out of the loop to avoid it being called upon each iteration. Also `++idx` is considered as more efficient.

```cpp
std::size_t m_vec_size = m_vec.size();
for (std::size_t idx = 0; idx < m_vec_size; ++idx) {}
```